### PR TITLE
Images by tag

### PR DESCRIPTION
**Pull request by swarajpatel**
PR for Issue 6

Card: https://trello.com/c/RI14LJ2l/151-issue-6-images-by-tag

### CODE REVIEW

**Comment by swarajpatel** [comment]
[Issue 6: Images by Tag](https://trello.com/c/RI14LJ2l/151-issue-6-images-by-tag)

File: app/views/images/index.html.erb [hunk 1]
```diff
@@ -3,7 +3,11 @@
   <div>
     <%= image_tag(image.url, class: 'display_image') %>
     <% if image.tag_list.present? %>
-      <p class="image_tags"> Tags: <%= image.tag_list %> </p>
+      <p class="image_tags">
+        Tags:<% image.tag_list.each do |tag| %>
+            <a href = <%= images_path(filter: tag) %>> <%= tag %></a>
```
Comment (fanniey) [inline app/views/images/index.html.erb]: the desired syntax should be `<a href=<%= images_path(filter: tag) %>><%= tag %></a>`. Eliminate the three extra spaces.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -70,7 +70,7 @@ def test_index_with_tags
     Image.create!(url: 'http://some-image.com/2.jpg')
     get images_path
     assert_response :ok
-    assert_select '.image_tags', 'Tags: tag1, tag2'
+    assert_select 'a[href=?]', '/images?filter=tag1'
```
Comment (fanniey) [inline test/controllers/images_controller_test.rb]: Can we also make sure the link for `tag2` exists and there is no link on `image2`? 

So basically, we want to make sure the view should render correctly with or without tags, and each tags are showing in the correct location.

File: test/controllers/images_controller_test.rb [hunk 3]
```diff
@@ -73,6 +73,14 @@ def test_index_with_tags
...
   end
 
+  def test_index_with_filter
+    Image.create!(url: 'http://some-image.com/1.jpg', tag_list: 'tag1')
+    Image.create!(url: 'http://some-image.com/2.jpg', tag_list: 'tag2')
+    get images_path(filter: 'tag1')
+    assert_response :ok
+    assert_select 'a[href=?]', '/images?filter=tag1'
+  end
+
```
Comment (fanniey) [inline test/controllers/images_controller_test.rb]: One thing I forgot to mention is that we prefer some extra lines in between the setup and the assertions. 

My preference is this: 
```
 	  def some_test
        test_setup
    
        exec_target_method
    
        setup_expectation_for_assertions

        assertions
    end
```

To apply to your test, it should look like: 
```
 	  def test_index_with_filter
        Image.create!(url: 'http://some-image.com/1.jpg', tag_list: 'tag1')
        Image.create!(url: 'http://some-image.com/2.jpg', tag_list: 'tag2')
    
        get images_path(filter: 'tag1')
    
        assert_response :ok
        assert_select 'a[href=?]', '/images?filter=tag1'
    end
```

File: test/controllers/images_controller_test.rb [hunk 4]
```diff
@@ -68,6 +74,7 @@ def test_index
   def test_index_with_tags
     Image.create!(url: 'http://some-image.com/1.jpg', tag_list: 'tag1, tag2')
     Image.create!(url: 'http://some-image.com/2.jpg')
+
     get images_path
```
Comment (fanniey) [inline test/controllers/images_controller_test.rb]: Can we add another new line here.

**Pull request merged**